azure/azure.go

```
@@ -46,11 +58,112 @@ var (
 
 type azureOps struct {
 	cloudops.Compute
-	instance          string
-	resourceGroupName string
-	disksClient       *compute.DisksClient
-	vmsClient         vmsClient
-	snapshotsClient   *compute.SnapshotsClient
+	instance           string
+	resourceGroupName  string
+	managedClusterName string
+	agentPoolName      string
+	disksClient        *compute.DisksClient
+	vmsClient          vmsClient
+	snapshotsClient    *compute.SnapshotsClient
+	agentPoolsClient   *containerservice.AgentPoolsClient
+}
+
+// NewAzureClient initializes cloudops driver for azure based on environment
+// variables or based on instance metadata info available inside Azure VM
+func NewAzureClient() (cloudops.Ops, error) {
+
+	if onAzure() {
+		logrus.Info("Running on Azure IaaS VM")
+
+		client := &http.Client{}
+
+		req, _ := http.NewRequest("GET", metadataAPIEndpoint, nil)
```

nit: don't ignore err here